### PR TITLE
Updated enable_http_endpoint references to show when it is properly v…

### LINKED ISSUE #35370: [Docs]: aws_rds_cluster: Incorrect description for enable_http_endpoint Argument
### Documentation Link

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/rds_cluster#enable_http_endpoint

### Description

The description of that argument says that it is only valid, when engine_mode is set to "serverless", but this is not true anymore. Serverless v2 now also supports Data API for postgresql DBs and neither this module nor the AWS API stops one from doing so, which means code with enable_http_endpoint on engine_mode "provisioned" (= Serverless v2) does work. It is just the documentation which is wrong.

It should say: (Optional) Enable HTTP endpoint (data API). Only valid when engine_mode is set to serverless or (if engine is set to aurora-postgresql) to provisioned.

### References

https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/Concepts.Aurora_Fea_Regions_DB-eng.Feature.Data_API.html

### Would you like to implement a fix?

No

### PR DESCRIPTION
**Pull request by jackedgar24**
### Description
Updated wording on enable_http_endpoint to described an allowed value(provisioned) when the engine is set as aurora-postgresql

### Relations

Closes #35370

### CODE REVIEW

File: website/docs/r/rds_cluster.html.markdown [hunk 1]
```diff
@@ -229,7 +229,7 @@ This argument supports the following arguments:
   The database can't be deleted when this value is set to `true`.
   The default is `false`.
 * `enable_global_write_forwarding` - (Optional) Whether cluster should forward writes to an associated global cluster. Applied to secondary clusters to enable them to forward writes to an [`aws_rds_global_cluster`](/docs/providers/aws/r/rds_global_cluster.html)'s primary cluster. See the [Aurora Userguide documentation](https://docs.aws.amazon.com/AmazonRDS/latest/AuroraUserGuide/aurora-global-database-write-forwarding.html) for more information.
-* `enable_http_endpoint` - (Optional) Enable HTTP endpoint (data API). Only valid when `engine_mode` is set to `serverless`.
+* `enable_http_endpoint` - (Optional) Enable HTTP endpoint (data API). Only valid when `engine_mode` is set to `serverless` or (if engine is set to aurora-postgresql) to `provisioned`.
```
Comment (justinretzolk) [inline website/docs/r/rds_cluster.html.markdown:232]: While this suggestion is wordier, I think it untangles the sentence a bit.

```suggestion
* `enable_http_endpoint` - (Optional) Enable HTTP endpoint (data API). Only valid when `engine_mode` is set to `serverless` or when `engine` is set to `aurora-postgresql` and `engine_mode` is set to `provisioned`.
```

**Review by justinretzolk — CHANGES_REQUESTED** [review]
Thank you for taking the time to raise this! Made a slight wording suggestion here. You'll also want to back out the change to the cdktf/python version of the documentation as well; those files are generated from the markdown versions 🙂

File: website/docs/r/rds_cluster.html.markdown (hunk 1 above)
Comment (jackedgar24) [reply website/docs/r/rds_cluster.html.markdown]: Thanks for the suggestion! I backed out the change to the cdktf/python version and updated the wording as asked.

**Comment by team-tf-cdk** [comment]
Superseded by newer PR.

**Comment by jritsema** [comment]
> Superseded by newer PR.

Which PR resolves this?  Can you provide link?

**Pull request closed without merging**